src/cargo/ops/cargo_add/mod.rs

```diff
@@ -943,56 +943,57 @@ fn print_dep_table_msg(shell: &mut Shell, dep: &DependencyUI) -> CargoResult<()>
     if matches!(shell.verbosity(), crate::core::shell::Verbosity::Quiet) {
         return Ok(());
     }
+
     let (activated, deactivated) = dep.features();
-    if !activated.is_empty() || !deactivated.is_empty() {
-        let prefix = format!("{:>13}", " ");
-        let suffix = format_feature_version_suffix(&dep);
-
-        shell.write_stderr(format_args!("{}Features{}:\n", prefix, suffix), &style::NOP)?;
-
-        const MAX_FEATURE_PRINTS: usize = 50;
-
-        let mut activated_printed = 0;
-        let total_activated = activated.len();
-        for feat in activated {
-            if activated_printed >= MAX_FEATURE_PRINTS {
-                let remaining = total_activated - activated_printed;
-                shell.write_stderr(
-                    format_args!("{prefix}... {remaining} more activated features\n"),
-                    &style::NOP,
-                )?;
-                break;
-            }
+    if activated.is_empty() && deactivated.is_empty() {
+        return Ok(());
+    }
```

Comment:
While I generally prefer guard statements / early returns,. I don't think its appropriate here as we could add more formatting after features and could easily miss his early return and miss it in the output that the extra information is dropped.

Reply:
I'd argue in this case it serves a similar purpose as lines 943~945:

```rs
    if matches!(shell.verbosity(), crate::core::shell::Verbosity::Quiet) {
        return Ok(());
    }
```

To me, these lines are saying: "If we don't need to display anything, stop here and leave". What are the reasons we don't display anything?
- The shell verbosity is set to quiet
- There are no features to display

In this case, I'd really say this is the intended logic.

But then again, you could easily argue the opposite: "if there are features to display, display them". I'd still lean towards keeping this section as is, but I don't have a strong enough opinion about it to not change it if you prefer the late return.

> and could easily miss his early return and miss it in the output that the extra information is dropped.

I think the tests would catch these cases quite easily.

Reply:
> I'd argue in this case it serves a similar purpose as lines 943~945:

That block says "everything after this should be bypassed for quiet".  That is very different from what your code is doing which can early return before feature unrelated stuff.

I feel fairly strongly that an early return here is inappropriate and you are laying a trap for future cargo contributors.

> I think the tests would catch these cases quite easily.

It would require people writing new tests for there could that hit it.

Reply:
That's understandable, I've reverted this change